File: GoldenForum.Service/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace GoldenForum.Service.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string UserId { get; set; }
        public int ForumId { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public DateTime PostedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        [ForeignKey("ForumId")]
        public virtual Forum Forum { get; set; }
        public virtual ICollection<Reply> Replies { get; set; }
    }
}

File: GoldenForum.Web/src/app/models/post.ts
import { Reply } from './reply';
import { User } from './user';

export class Post {
    id: number;
    title: string;
    body: string;
    repliesCount: number;
    postedAt: Date;
    modifiedAt?: Date;
    isDeleted: boolean;
    type: string;

    author: User = new User();
    replies: Reply[] = [];
}